ee_tests/src/specs/quickstart_cheterminal.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { browser, Key, ExpectedConditions as until, $, $$ } from 'protractor';
+import * as support from './support';
+import { Quickstart } from './support/quickstart';
+import { TextInput, Button } from './ui';
+import { LandingPage } from './page_objects/landing.page';
+import { SpaceDashboardPage } from './page_objects/space_dashboard.page';
+import { SpaceChePage } from './page_objects/space_che.page';
+import { SpaceCheWorkspacePage } from './page_objects/space_cheworkspace.page';
+import { MainDashboardPage } from './page_objects/main_dashboard.page';
+
+let globalSpaceName: string;
+
+/* Tests to verify Che in OSIO */
+
+describe('Creating new quickstart in OSIO', () => {
+  let dashboardPage: MainDashboardPage;
+
+  beforeEach(async () => {
+    await support.desktopTestSetup();
+    let login = new support.LoginInteraction();
+    dashboardPage = await login.run();
+
+    let userProfilePage = await dashboardPage.gotoUserProfile();
+    support.debug(">>> Go to user's Profile Page - OK");
+    support.debug('>>> Go to Edit Profile Page');
+    let editProfilePage = await userProfilePage.gotoEditProfile();
+    support.debug('>>> Go to Edit Profile Page - OK');
+    support.debug('>>> Go to Reset Env Page');
+    let cleanupEnvPage = await editProfilePage.gotoResetEnvironment();
+    support.debug('>>> Go to Reset Env Page - OK');
+
+    await cleanupEnvPage.cleanup(browser.params.login.user);
```

Comment:
This is too dangerous... it either has to wait for account to be in 'ready' state again of simply assume that the software under test is ready to be used.

Reply:
This one requires some explanation as it really is a short-term fix to multiple issues. 

*  Ideally, every test would reset its environment before running. Resetting the environment enables a test to start with a known (and clean) environment. Also, since our test accounts are all provisioned in the OSO free tier, they are configured with limited resources. Running multiple tests can result in OSO quotas being exceeded.
* This is not possible at present for multiple reasons - First, resetting the environment is time consuming. Second, the time required is variable. Third, the reset can fail and leave the environment in a state where it must be reset again (perhaps more than once). Fourth and finally, the reset can appear to be successful, but the Jenkins pod is not started.
* The long term solution is to provide the user with a safe and quick way to "clean up" their OSO resources. This clean up feature does not yet exist. When it is created (after the H train?), the feature to reset the user's environment will be removed or hidden from the user.

Given these^ restrictions, the best short-term approach that I have

Reply:
I was not able to successfully run your test and understand why this env reset is present and what problems it has. I would prefer not merging tests that depend on unstable features but I'll leave you to decide if to merge or not.

Reply:
Isn't it the only valid reason for tests to exist - to fail? Just my .2$

Reply:
YES! I agree 100% 

Can we talk about a better way to clean'reset a user's env? The current workaround is just that; a workaround.

But - we have to find a way to contend with and workaround these issues:

Issues related to Environment Reset:
* https://github.com/openshiftio/openshift.io/issues/2395, Intermittent PVC errors
* https://github.com/openshiftio/openshift.io/issues/2390, Delays in Jenkins starting
* https://github.com/openshiftio/openshift.io/issues/2232, Delays in Jenkins starting
* https://github.com/openshiftio/openshift.io/issues/2200, Infinite loop after reset
* https://github.com/openshiftio/openshift.io/issues/1637, Intermittent reset failures
* https://github.com/openshiftio/openshift.io/issues/945, After reset, very slow reponse